apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,29 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  ContentChild,
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
+import { CardImageDirective } from './card-image.directive';
+import { CardItemDirective } from './card-item.directive';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
+  host: {
+    class: 'border-2 border-black rounded-md p-4 w-fit flex flex-col gap-3',
+  },
 })
 export class CardComponent {
   @Input() list: any[] | null = null;
-  @Input() type!: CardType;
-  @Input() customClass = '';
 
-  CardType = CardType;
+  @Output() add = new EventEmitter<void>();
 
-  constructor(
-    private teacherStore: TeacherStore,
-    private studentStore: StudentStore
-  ) {}
-
-  addNewItem() {
-    if (this.type === CardType.TEACHER) {
-      this.teacherStore.addOne(randTeacher());
-    } else if (this.type === CardType.STUDENT) {
-      this.studentStore.addOne(randStudent());
-    }
-  }
+  @ContentChild(CardImageDirective) cardImage!: CardImageDirective;
+  @ContentChild(CardItemDirective) cardItem!: CardItemDirective;
```

Comment:
Using a directive is the best way to do it. nice job 👍

If you want, you can use the property read of ContentChild to create a TemplateRef, instead of using the contructor of the directive, but both way work perfectly